lib/Runtime/Library/WebAssemblyMemory.cpp

```diff
@@ -149,24 +181,37 @@ WebAssemblyMemory::GrowInternal(uint32 deltaPages)
         return -1;
     }
 
-    WebAssemblyArrayBuffer * newBuffer = nullptr;
-    JavascriptExceptionObject* caughtExceptionObject = nullptr;
-    try
+    AssertOrFailFast(m_buffer->IsWebAssemblyArrayBuffer());
+#ifdef ENABLE_WASM_THREADS
+    if (m_buffer->IsSharedArrayBuffer())
     {
-        newBuffer = m_buffer->GrowMemory(newBytes);
-        if (newBuffer == nullptr)
+        AssertOrFailFast(Wasm::Threads::IsEnabled());
+        if (!((WebAssemblySharedArrayBuffer*)PointerValue(m_buffer))->GrowMemory(newBytes))
```

Comment:
doing PointerValue strikes me as a bad practice, maybe we should be casting to Field(WebAssemblySharedArrayBuffer*)?